File: testdata/jennies/builders/discriminator_without_option/JavaBuilder/discriminator_without_option/ShowFieldOptionBuilder.java
package discriminator_without_option;


public class ShowFieldOptionBuilder implements cog.Builder<ShowFieldOption> {
    protected final ShowFieldOption internal;
    
    public ShowFieldOptionBuilder() {
        this.internal = new ShowFieldOption();
    }
    public ShowFieldOptionBuilder field(AnEnum field) {
        this.internal.field = field;
        return this;
    }
    
    public ShowFieldOptionBuilder text(String text) {
        this.internal.text = text;
        return this;
    }
    public ShowFieldOption build() {
        return this.internal;
    }
}

File: testdata/jennies/builders/discriminator_without_option/PHPBuilder/src/DiscriminatorWithoutOption/ShowFieldOptionBuilder.php
<?php

namespace Grafana\Foundation\DiscriminatorWithoutOption;

/**
 * @implements \Grafana\Foundation\Cog\Builder<\Grafana\Foundation\DiscriminatorWithoutOption\ShowFieldOption>
 */
class ShowFieldOptionBuilder implements \Grafana\Foundation\Cog\Builder
{
    protected \Grafana\Foundation\DiscriminatorWithoutOption\ShowFieldOption $internal;

    public function __construct()
    {
    	$this->internal = new \Grafana\Foundation\DiscriminatorWithoutOption\ShowFieldOption();
    }

    /**
     * Builds the object.
     * @return \Grafana\Foundation\DiscriminatorWithoutOption\ShowFieldOption
     */
    public function build()
    {
        return $this->internal;
    }

    public function field(\Grafana\Foundation\DiscriminatorWithoutOption\AnEnum $field): static
    {
        $this->internal->field = $field;
    
        return $this;
    }

    public function text(string $text): static
    {
        $this->internal->text = $text;
    
        return $this;
    }

}
